src/gordon_gcp/clients/http.py

```diff
@@ -114,23 +113,28 @@ def __init__(self, auth_client=None, session=None):
 
         req_headers = headers or {}
         req_headers.update(_utils.DEFAULT_REQUEST_HEADERS)
-
-        await self.set_valid_token()
-        req_headers.update(
-            {'Authorization': f'Bearer {self._auth_client.token}'}
-        )
-
         req_kwargs = {
-            'params': params,
-            'data': body,
-            'headers': req_headers,
-        }
-        logging.debug(_utils.REQ_LOG_FMT.format(method=method.upper(), url=url))
-        async with self._session.request(method, url, **req_kwargs) as resp:
+                'params': params,
+                'headers': req_headers,
+            }
+
         if data:
             req_kwargs['data'] = data
         if json:
             req_kwargs['json'] = json
+
+        if token_refresh_attempts:
+            if not await self.valid_token_set():
+                await self._auth_client.refresh_token()
+                req_headers.update(
```

Comment:
If the authorization header is only added to the headers when there is no valid token set (eg. first request), won't further requests be missing the Authorization header?

Reply:
Talked IRL - yes this needs to be moved below and set every time! Thanks for pointing this out.